### ISSUE TITLE
Error at startup on Insteon Hub Pro

### ISSUE THREAD

**Issue by ctgreybeard** [kuestess/homebridge-platform-insteonlocal#55]
I have my Hub Pro reconfigured and running off the SD card OK. I am having problems, though, getting the UART enabled. I don't see the symlinks in /dev that were described in another issue and HomeBridge complains that it can't set the baud rate on the UART and, of course, none of my Insteon devices are showing correctly. They **do** appear in the Home app but are all `Off` when they are not really.

I think I made the correct changes to /boot/uEnv.txt and the startup messages seem to indicate that it's trying to do what I asked. Here's what I see:

```
[    1.591926] bone_capemgr bone_capemgr: Baseboard: 'A335BNLT,000B,3414SALB0015'
[    1.591957] bone_capemgr bone_capemgr: compatible-baseboard=ti,beaglebone-black - #slots=4
[    1.620557] bone_capemgr bone_capemgr: slot #0: No cape found
[    1.647500] bone_capemgr bone_capemgr: slot #1: No cape found
[    1.676022] bone_capemgr bone_capemgr: slot #2: No cape found
[    1.704585] bone_capemgr bone_capemgr: slot #3: No cape found
[    1.704606] bone_capemgr bone_capemgr: enabled_partno PARTNO 'BB-UART1' VER 'N/A' PR '0'
[    1.704643] bone_capemgr bone_capemgr: slot #4: override
[    1.704657] bone_capemgr bone_capemgr: Using override eeprom data at slot 4
[    1.704669] bone_capemgr bone_capemgr: slot #4: 'Override Board Name,00A0,Override Manuf,BB-UART1'
[    1.704748] bone_capemgr bone_capemgr: enabled_partno PARTNO 'BB-UART2' VER 'N/A' PR '0'
[    1.704758] bone_capemgr bone_capemgr: slot #5: override
[    1.704767] bone_capemgr bone_capemgr: Using override eeprom data at slot 5
[    1.704778] bone_capemgr bone_capemgr: slot #5: 'Override Board Name,00A0,Override Manuf,BB-UART2'
[    1.704826] bone_capemgr bone_capemgr: enabled_partno PARTNO 'BB-UART4' VER 'N/A' PR '0'
[    1.704836] bone_capemgr bone_capemgr: slot #6: override
[    1.704844] bone_capemgr bone_capemgr: Using override eeprom data at slot 6
[    1.704855] bone_capemgr bone_capemgr: slot #6: 'Override Board Name,00A0,Override Manuf,BB-UART4'
[    1.704916] bone_capemgr bone_capemgr: enabled_partno PARTNO 'BB-UART5' VER 'N/A' PR '0'
[    1.704926] bone_capemgr bone_capemgr: slot #7: override
[    1.704934] bone_capemgr bone_capemgr: Using override eeprom data at slot 7
[    1.704945] bone_capemgr bone_capemgr: slot #7: 'Override Board Name,00A0,Override Manuf,BB-UART5'
[    1.705618] bone_capemgr bone_capemgr: initialized OK.
[    1.723651] bone_capemgr bone_capemgr: loader: failed to load slot-7 BB-UART5:00A0 (prio 0)
[    1.732260] bone_capemgr bone_capemgr: loader: failed to load slot-5 BB-UART2:00A0 (prio 0)
[    1.733686] cpufreq: cpufreq_online: CPU0: Running at unlisted freq: 1000000 KHz
[    1.733725] cpu cpu0: dev_pm_opp_set_rate: failed to find current OPP for freq 1000000000 (-34)
[    1.734099] bone_capemgr bone_capemgr: loader: failed to load slot-6 BB-UART4:00A0 (prio 0)
[    1.734474] bone_capemgr bone_capemgr: loader: failed to load slot-4 BB-UART1:00A0 (prio 0)
[    1.773735] cpufreq: cpufreq_online: CPU0: Unlisted initial frequency changed to: 600000 KHz
[    1.775955] PM: bootloader does not support rtc-only!
[    1.778243] omap_rtc 44e3e000.rtc: setting system clock to 2019-02-17 00:57:07 UTC (1550365027)
```

The errors are `failed to load slot-7 BB-UART5:00A0` and three others.

The lines in uEnv.txt are:

```
uboot_overlay_addr0=/lib/firmware/BB-UART1-00A0.dtbo
uboot_overlay_addr1=/lib/firmware/BB-UART2-00A0.dtbo
uboot_overlay_addr2=/lib/firmware/BB-UART4-00A0.dtbo
uboot_overlay_addr3=/lib/firmware/BB-UART5-00A0.dtbo
```

Those are placed near the top right after the commented similar examples, and:

```
cape_disable=bone_capemgr.disable_partno=BB-BONELT-HDMI,BB-BONELT-HDMIN
cape_enable=bone_capemgr.enable_partno=BB-UART1,BB-UART2,BB-UART4,BB-UART5
```

are the last two lines. I did try uncommenting:

```
disable_uboot_overlay_video=1
disable_uboot_overlay_audio=1
disable_uboot_overlay_wireless=1
```

which didn't affect the problem.

It also doesn't seem to matter whether I boot with or without `S2` held down. Even when I don't use `S2` it appears to boot from the floppy. I checked /boot/uEnv.txt on the eMMC (/dev/mmcblk1p1) partition and it's nothing at all like what is on the SD card.

**Comment by rasod** [comment]
I have noticed that if you don’t hold down S2 on boot it will still use the SD image but the serial port will only be enabled when using the S2 button.

Could be that enabling the port has changed since I originally wrote the instructions.

**Comment by rasod** [comment]
This thread seems to indicated that enabling the UARTS has changed. 
https://stackoverflow.com/a/43175619

**Comment by ctgreybeard** [comment]
Looking on the eMMC which the HubPro runs under I don't see anything at all that enables the UARTS, either in uEnv.txt or capemgr. How are they accessing them then? Is it possible that they are already enabled (how can I test that?) and we just need to point the code to a different tty?

**Comment by shereefalkady** [comment]
@rasod I just reinstalled and used the 2243 using the same instructions from last year everything still works like a charm ...

One thing I think noticed however ... your link.js only does 1 way links ? It doesn’t crosslink devices to report back?

**Comment by ctgreybeard** [comment]
I don't understand the question about link.js.

I wonder what's different about my setup ... 

I looked on the original code and found their slserver has this:

`startup.sh:     cmd="./slserver -config:../../config/$config_version/insteon.xml -settings:../../settings.xml -serial:/dev/ttyO4"`

And they have nothing in uEnv.txt to enable any UART.

**Comment by ctgreybeard** [comment]
@shereefalkady What does your uEnv.txt look like? Enabling the UARTs seems to be the sticking point for my system. Also, what base did you use to build the SD card? I followed the instructions I think. Because the SD card boots I assume that I got it correct.

**Comment by ctgreybeard** [comment]
Can you run dmesg and tell me what a "good" UART load looks like? I see this:

```
[    1.591920] bone_capemgr bone_capemgr: Baseboard: 'A335BNLT,000B,3414SALB0015'
[    1.591949] bone_capemgr bone_capemgr: compatible-baseboard=ti,beaglebone-black - #slots=4
[    1.620555] bone_capemgr bone_capemgr: slot #0: No cape found
[    1.647486] bone_capemgr bone_capemgr: slot #1: No cape found
[    1.676001] bone_capemgr bone_capemgr: slot #2: No cape found
[    1.704567] bone_capemgr bone_capemgr: slot #3: No cape found
[    1.704588] bone_capemgr bone_capemgr: enabled_partno PARTNO 'BB-UART1' VER 'N/A' PR '0'
[    1.704625] bone_capemgr bone_capemgr: slot #4: override
[    1.704639] bone_capemgr bone_capemgr: Using override eeprom data at slot 4
[    1.704651] bone_capemgr bone_capemgr: slot #4: 'Override Board Name,00A0,Override Manuf,BB-UART1'
[    1.704731] bone_capemgr bone_capemgr: enabled_partno PARTNO 'BB-UART2' VER 'N/A' PR '0'
[    1.704740] bone_capemgr bone_capemgr: slot #5: override
[    1.704750] bone_capemgr bone_capemgr: Using override eeprom data at slot 5
[    1.704760] bone_capemgr bone_capemgr: slot #5: 'Override Board Name,00A0,Override Manuf,BB-UART2'
[    1.704810] bone_capemgr bone_capemgr: enabled_partno PARTNO 'BB-UART4' VER 'N/A' PR '0'
[    1.704819] bone_capemgr bone_capemgr: slot #6: override
[    1.704828] bone_capemgr bone_capemgr: Using override eeprom data at slot 6
[    1.704839] bone_capemgr bone_capemgr: slot #6: 'Override Board Name,00A0,Override Manuf,BB-UART4'
[    1.704898] bone_capemgr bone_capemgr: enabled_partno PARTNO 'BB-UART5' VER 'N/A' PR '0'
[    1.704908] bone_capemgr bone_capemgr: slot #7: override
[    1.704917] bone_capemgr bone_capemgr: Using override eeprom data at slot 7
[    1.704928] bone_capemgr bone_capemgr: slot #7: 'Override Board Name,00A0,Override Manuf,BB-UART5'
[    1.705600] bone_capemgr bone_capemgr: initialized OK.
[    1.724137] bone_capemgr bone_capemgr: loader: failed to load slot-7 BB-UART5:00A0 (prio 0)
[    1.732715] bone_capemgr bone_capemgr: loader: failed to load slot-5 BB-UART2:00A0 (prio 0)
[    1.734160] cpufreq: cpufreq_online: CPU0: Running at unlisted freq: 1000000 KHz
[    1.734199] cpu cpu0: dev_pm_opp_set_rate: failed to find current OPP for freq 1000000000 (-34)
[    1.734599] bone_capemgr bone_capemgr: loader: failed to load slot-4 BB-UART1:00A0 (prio 0)
[    1.734925] bone_capemgr bone_capemgr: loader: failed to load slot-6 BB-UART4:00A0 (prio 0)
```

**Comment by shereefalkady** [comment]
@ctgreybeard are you trying to autorun it  ?  might be permissions?

here is the uEnv:


login as: debian
debian@192.168.86.22's password:

The programs included with the Debian GNU/Linux system are free software;
the exact distribution terms for each program are described in the
individual files in /usr/share/doc/*/copyright.

Debian GNU/Linux comes with ABSOLUTELY NO WARRANTY, to the extent
permitted by applicable law.
Last login: Fri Feb 15 06:00:28 2019 from 192.168.86.42
debian@beaglebone:~$ sudo nano /boot/uEnv.txt
[sudo] password for debian:
debian@beaglebone:~$
debian@beaglebone:~$ nano /boot/uEnv.txt
  GNU nano 2.7.4                                                 File: /boot/uEnv.txt

#Docs: http://elinux.org/Beagleboard:U-boot_partitioning_layout_2.0

uname_r=4.14.71-ti-r80
#uuid=
#dtb=

cape_disable=bone_capemgr.disable_partno=BB-BONELT-HDMI,BB-BONELT-HDMIN
cape_enable=bone_capemgr.enable_partno=BB-UART1,BB-UART2,BB-UART4,BB-UART5

###U-Boot Overlays###
###Documentation: http://elinux.org/Beagleboard:BeagleBoneBlack_Debian#U-Boot_Overlays
###Master Enable
enable_uboot_overlays=1
###
###Overide capes with eeprom
uboot_overlay_addr0=/lib/firmware/BB-UART1-00A0.dtbo
uboot_overlay_addr1=/lib/firmware/BB-UART2-00A0.dtbo
uboot_overlay_addr2=/lib/firmware/BB-UART4-00A0.dtbo
uboot_overlay_addr3=/lib/firmware/BB-UART5-00A0.dtbo
###
###Additional custom capes
#uboot_overlay_addr4=/lib/firmware/<file4>.dtbo
#uboot_overlay_addr5=/lib/firmware/<file5>.dtbo
#uboot_overlay_addr6=/lib/firmware/<file6>.dtbo
#uboot_overlay_addr7=/lib/firmware/<file7>.dtbo
###
###Custom Cape
#dtb_overlay=/lib/firmware/<file8>.dtbo
###
###Disable auto loading of virtual capes (emmc/video/wireless/adc)
#disable_uboot_overlay_emmc=1
#disable_uboot_overlay_video=1
#disable_uboot_overlay_audio=1
#disable_uboot_overlay_wireless=1
#disable_uboot_overlay_adc=1
###
###PRUSS OPTIONS
###pru_rproc (4.4.x-ti kernel)
#uboot_overlay_pru=/lib/firmware/AM335X-PRU-RPROC-4-4-TI-00A0.dtbo
###pru_rproc (4.14.x-ti kernel)
#uboot_overlay_pru=/lib/firmware/AM335X-PRU-RPROC-4-14-TI-00A0.dtbo
###pru_uio (4.4.x-ti, 4.14.x-ti & mainline/bone kernel)
#uboot_overlay_pru=/lib/firmware/AM335X-PRU-UIO-00A0.dtbo
###
###Cape Universal Enable
enable_uboot_cape_universal=1
###
###Debug: disable uboot autoload of Cape
#disable_uboot_overlay_addr0=1
#disable_uboot_overlay_addr1=1
#disable_uboot_overlay_addr2=1
#disable_uboot_overlay_addr3=1
###
###U-Boot fdt tweaks... (60000 = 384KB)
#uboot_fdt_buffer=0x60000
###U-Boot Overlays###

**Comment by shereefalkady** [comment]
@ctgreybeard if you are getting this error: "Cannot open /dev/ttyS0: Permission denied"

let me know - i have a fix for that..

**Comment by ctgreybeard** [comment]
Well, I have solved the UART problem. Thanks, all, for your help. It turns out that I had, in my ignorance, created a uEnv.txt in / on the SD card. This was taking control at boot time.

I didn't dig further but I believe that because it didn't have the line
```
enable_uboot_overlays=1
```
in it then nothing would load. What I had done was to copy /bbb-uEnv.txt as /uEnv.txt and updated it but that file isn't very useful even though it appears to be the one to use.

Once I deleted the errant /uEnv.txt the one /boot/uEnv.txt was recognized and the overlays load.

I now have communication with the controller but it still isn't working correctly. I will play with it some more to see what I can find out.
